File: Graph/yeonchon/1260.py
import sys
from collections import deque # 양쪽 끝에서 추가 제거 가능

def dfs(graph, start):
    visited = set()
    stack = [start]
    
    while stack:
        node = stack.pop()
        
        if node not in visited:
            visited.add(node)
            print(node, end=" ")
            
            # 정점 번호가 작은 것을 먼저 방문
            for nextNode in reversed(graph[node]):
                stack.append(nextNode)
    print()        

def bfs(graph, start):
    visited = set()
    visited.add(start)
    queue = deque([start])
    
    while queue:
        node = queue.popleft() # 맨 앞 값 제거 후 반환
        print(node, end=" ")
        
        for nextNode in graph[node]:
            if nextNode not in visited:
                visited.add(nextNode) # 방문 set 에 추가
                queue.append(nextNode) # 맨 뒤에 추가


v, e, start = map(int, sys.stdin.readline().strip().split())

# 그래프 초기화
graph = {i: [] for i in range(1, v + 1)}

for i in range(e):
    
    a, b = map(int, sys.stdin.readline().strip().split())
    
    if a not in graph:
        graph[a] = list()
    if b not in graph:
        graph[b] = list()
        
    graph[a].append(b)
    graph[b].append(a)

for edgeList in graph.values():
    edgeList.sort()

dfs(graph, start)
bfs(graph, start)

File: Graph/jichon/1260.js
const fs = require("fs");
const input = fs.readFileSync("./input.txt").toString().trim().split("\n");

function buildGraph(N, edges) {
  const graph = Array.from({ length: N + 1 }, () => []);

  for (const [a, b] of edges) {
    graph[a].push(b);
    graph[b].push(a);
  }
  graph.forEach((neighbors) => neighbors.sort((a, b) => a - b));

  return graph;
}

function dfs(graph, start, visited) {
  visited[start] = true;
  process.stdout.write(start + " "); //한줄에 출력해야됨...

  for (const neighbor of graph[start]) {
    if (!visited[neighbor]) {
      dfs(graph, neighbor, visited);
    }
  }
}

function bfs(graph, start) {
  const queue = [start];
  const visited = Array(graph.length).fill(false);
  visited[start] = true;

  while (queue.length) {
    const node = queue.shift();
    process.stdout.write(node + " ");

    //너비 우선 -> 그 같은 정점이랑 연결된 간선들 먼저 다 싹 봐야되니까.
    for (const neighbor of graph[node]) {
      if (!visited[neighbor]) {
        visited[neighbor] = true;
        queue.push(neighbor);
      }
    }
  }
}

const solution = () => {
  let [N, M, V] = input[0].split(" ").map(Number);
  const edges = input.slice(1).map((line) => line.split(" ").map(Number));

  const graph = buildGraph(N, edges);

  let visited = Array(N + 1).fill(false);
  dfs(graph, V, visited);
  console.log();
  bfs(graph, V);
};

solution();
